bufferevent_sock.c

```diff
@@ -498,6 +498,41 @@ bufferevent_connect_getaddrinfo_cb(int result, struct evutil_addrinfo *ai,
 }
 
 int
+bufferevent_socket_set_src_address(struct bufferevent *bev,
+    const struct sockaddr *addr, int socklen)
+{
+	struct bufferevent_private *bufev_p =
+	    EVUTIL_UPCAST(bev, struct bufferevent_private, bev);
+
+	evutil_socket_t fd;
+	int r = 0;
+	int result = -1;
+
+	bufferevent_incref_and_lock_(bev);
+
+	if (!bufev_p || !addr)
+		goto done;
+
+	/* do we need to check to see if the socket is connecting/connected
+	 * already?
```

Comment:
Yeah, I wanted to check specifically before the operation, but we can proceed and as you say, the value of `errno` will give the reason it failed.